src/mono/mono/metadata/icall.c

```diff
@@ -2886,6 +2886,32 @@ ves_icall_RuntimeTypeHandle_IsByRefLike (MonoQCallTypeHandle type_handle, MonoEr
 	return !!m_class_is_byreflike (klass);
 }
 
+MonoBoolean
+ves_icall_RuntimeTypeHandle_IsUnmanagedFunctionPointer (MonoQCallTypeHandle type_handle, MonoError *error)
+{
+	MonoType *type = type_handle.type;
+	if (type->type == MONO_TYPE_FNPTR)
+		return type->data.method->call_convention == MONO_CALL_UNMANAGED_MD;
```

Comment:
I was a bit uncertain what would be the right thing to do because coreCLR does this: https://github.com/dotnet/runtime/blob/2b477aeeabc356bff43011fd75c57c725c2b4c77/src/coreclr/vm/runtimehandles.cpp#L912-L916 which seems like they are looking up only the case with default unmanaged calling convention (when cc is encoded as modopts). 

They mention `Type.IsUnmanagedFunctionPointer` in this section: https://github.com/steveharter/designs/blob/5c3594d882c336f651027a022a8c8c5941c627be/accepted/2022/FunctionPointers.md#normalization-of-calling-conventions

Reply:
We use sig->pinvoke to decide whenever a signature is native or not in many places in the runtime.
Its set at metadata.c:2670.

Reply:
Done.